packages/xo-server/src/xapi/index.js

```diff
@@ -1913,15 +1916,20 @@ export default class Xapi extends XapiBase {
           vdi: newVdi,
         })
       })
-      await this._deleteVdi(vdi)
+      await this._deleteVdi(vdi.$ref)
     }
   }
 
   // TODO: check whether the VDI is attached.
-  async _deleteVdi(vdi) {
-    log.debug(`Deleting VDI ${vdi.name_label}`)
+  async _deleteVdi(vdiRef) {
+    log.debug(`Deleting VDI ${vdiRef}`)
 
-    await this.call('VDI.destroy', vdi.$ref)
+    await this.call('VDI.destroy', vdiRef).catch(error => {
+      if (error.code !== 'HANDLE_INVALID') {
```

Comment:
Use `error?.code` to be safe in case error is `null` or `undefined` (should not happen but we never know with JS).